modules/cxf/src/main/java/org/mule/module/cxf/CxfInboundMessageProcessor.java

```diff
@@ -229,150 +235,192 @@ protected MuleEvent sendToDestination(MuleEvent event) throws MuleException, IOE
     {
         try
         {
-            final MessageImpl m = new MessageImpl();
-            final MuleMessage muleReqMsg = event.getMessage();
-            String method = muleReqMsg.getInboundProperty(HttpConnector.HTTP_METHOD_PROPERTY);
+            final Exchange exchange = new ExchangeImpl();
 
-            String ct = muleReqMsg.getInboundProperty(HttpConstants.HEADER_CONTENT_TYPE);
-            if (ct != null)
+            if (event.isAllowNonBlocking())
             {
-                m.put(Message.CONTENT_TYPE, ct);
+                final ReplyToHandler originalReplyToHandler = event.getReplyToHandler();
+
+                event = new DefaultMuleEvent(event, new ReplyToHandler()
+                {
+                    @Override
+                    public void processReplyTo(MuleEvent event, MuleMessage returnMessage, Object replyTo) throws MuleException
+                    {
+                        try
+                        {
+                            originalReplyToHandler.processReplyTo(event, returnMessage, replyTo);
+                        }
+                        catch (Exception e)
+                        {
+                            processExceptionReplyTo(new MessagingException(event, e), replyTo);
+                        }
+                    }
+
+                    @Override
+                    public void processExceptionReplyTo(MessagingException exception, Object replyTo)
+                    {
+                        originalReplyToHandler.processExceptionReplyTo(exception, replyTo);
+                    }
+                });
             }
 
-            String path = muleReqMsg.getInboundProperty(HttpConnector.HTTP_REQUEST_PATH_PROPERTY);
-            if (path == null)
+            MuleEvent responseEvent = sendThroughCxf(event, exchange);
+
+            if (responseEvent==null || !responseEvent.equals(NonBlockingVoidMuleEvent.getInstance()))
```

Comment:
the first part could be removed by using:
if (!NonBlockingVoidMuleEvent.getInstance().equals(responseEvent))
but I think that making explicit the null comparison is better here.